src/contracts/core/AVSDirectory.sol

```diff
@@ -261,6 +264,27 @@ contract AVSDirectory is
         _deregisterFromOperatorSets(msg.sender, operator, operatorSetIds);
     }
 
+
+    /**
+     * @notice Called by operators to set their allocation delay. 
+     * @param delay the allocation delay in seconds
+     * @dev msg.sender is assumed to be the operator
+     * @dev Fails if setting a delay would result in the ability to set an
+     *      an allocation that is active BEFORE the latest set allocation. 
+     */
+    function setAllocationDelay(uint32 delay) external {
+        require(
+            delegation.isOperator(msg.sender),
+            "avsDirectory.initializeAllocationDelay: operator not registered to EigenLayer yet"
+        );
+        require(
+            delay + block.timestamp >= latestAllocationEffectTimestamp[msg.sender],
```

Comment:
what does this condition get us?

Reply:
We want to make sure if you decrease your allocation delay, you can't insert a pending update into the checkpoints array with the timestamps not being ordered. Although this is not feasible anyways with 1 max pending allocation?

Reply:
Nvm: snapshot library has a ordering check in its `_insert` function for decreasing keys